src/main/java/xyz/nucleoid/plasmid/mixin/chat/PlayerEntityMixin.java

```diff
@@ -0,0 +1,24 @@
+package xyz.nucleoid.plasmid.mixin.chat;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import xyz.nucleoid.plasmid.game.event.GamePlayerEvents;
+import xyz.nucleoid.stimuli.Stimuli;
+
+@Mixin(PlayerEntity.class)
+public class PlayerEntityMixin {
+
+    @Inject(method = "getDisplayName", at = @At("TAIL"), cancellable = true)
+    private void callDisplayNameEvent(CallbackInfoReturnable<Text> cir) {
+        if (((Object) this) instanceof ServerPlayerEntity player) {
+            try (var invokers = Stimuli.select().forEntityAt(player, player.getBlockPos())) {
+                cir.setReturnValue(invokers.get(GamePlayerEvents.DISPLAY_NAME).onDisplayNameCreation(player, cir.getReturnValue(), cir.getReturnValue()));
```

Comment:
Can probably avoid calling this if it's not changed- should help for mod compatibility